integration/sql/src/lib.rs

```diff
@@ -0,0 +1,512 @@
+}
+
+fn parse_stmt(stmt: &Statement, context: &mut Context) -> Result<(), String> {
+    match stmt {
+        Statement::Query(query) => {
+            parse_query(query, context)?;
+            Ok(())
+        }
+        Statement::Insert {
+            or: _,
+            table_name,
+            columns: _,
+            overwrite: _,
+            source,
+            partitioned: _,
+            after_columns: _,
+            table: _,
+            on: _,
+        } => {
+            parse_query(source, context)?;
+            context.add_output(&table_name.to_string());
+            Ok(())
+        }
+        Statement::Merge {
+            table,
+            source,
+            alias,
+            on: _,
+            clauses: _,
+        } => {
+            let table_name = get_table_name_from_table_factor(table)?;
+            context.add_output(&table_name);
+            parse_setexpr(source, context)?;
+
+            if let Some(a) = alias {
+                context.add_table_alias(a);
+            }
+
+            Ok(())
+        }
+        Statement::CreateTable {
+            or_replace: _,
+            temporary: _,
+            external: _,
+            if_not_exists: _,
+            name,
+            columns: _,
+            constraints: _,
+            hive_distribution: _,
+            hive_formats: _,
+            table_properties: _,
+            with_options: _,
+            file_format: _,
+            location: _,
+            query,
+            without_rowid: _,
+            like,
+            engine: _,
+            default_charset: _,
+            collation: _,
+        } => {
+            if let Some(boxed_query) = query {
+                parse_query(boxed_query.as_ref(), context)?;
+            }
+            if let Some(like_table) = like {
+                context.add_input(&like_table.to_string());
+            }
+            context.add_output(&name.to_string());
+            Ok(())
+        }
+        _ => Ok(()),
+    }
+}
+
+pub fn parse_sql(
+    sql: &str,
+    dialect: Box<dyn CanonicalDialect>,
+    default_schema: Option<&str>,
+) -> Result<SqlMeta, String> {
+    let ast = match Parser::parse_sql(dialect.as_base(), sql) {
+        Ok(k) => k,
+        Err(e) => return Err(e.to_string()),
+    };
+
+    if ast.is_empty() {
+        return Err(String::from("Empty statement list"));
+    }
+
+    let mut context = Context::new(dialect, default_schema);
+    for stmt in ast {
+        parse_stmt(&stmt, &mut context)?;
+    }
+    Ok(SqlMeta::from(context))
+}
+
+// Parses SQL.
+#[pyfunction]
+fn parse(sql: &str, dialect: Option<&str>, default_schema: Option<&str>) -> PyResult<SqlMeta> {
+    let parser_dialect: Box<dyn CanonicalDialect> = if let Some(d) = dialect {
+        match d {
+            "bigquery" => Box::new(BigQueryDialect),
+            "snowflake" => Box::new(SnowflakeDialect),
+            "postgres" => Box::new(PostgreSqlDialect {}),
+            _ => Box::new(PostgreSqlDialect {}),
```

Comment:
>if it was just about how we parse the SQL, I don't think it would be a big deal

Yes, that's the case here. Our parser contract specify `SqlMeta` class, that contains lists (vectors in Rust, the same anyway) of `DbTableMeta` classes. The `DbTableMeta` class specifies names of database, schema, and name of table. 

Then, "consumer" of that information - like `PostgresExtractor` - is responsible to construct OpenLineage datasets from that information - since parser has no idea about context (Airflow, Spark, Great Expectations...) in which it operates. They also should provide dialect string to parser. Unfortunately, limitations of Python-Rust interoperability hit us here - we can't expose something like type-safe enum, so it's all string based.

Anyway, I've added more dialects - all that are supported by sqlparser-rs - and defaulted to `GenericDialect`.